src/v/utils/to_string.h

```diff
@@ -12,13 +12,29 @@
 #pragma once
 
 #include "seastarx.h"
+#include "ssx/sformat.h"
 
 #include <seastar/core/lowres_clock.hh>
 #include <seastar/core/print.hh>
 
 #include <optional>
 #include <ostream>
 
+inline ss::sstring human_readable_bytes(uint64_t bytes) {
+    constexpr std::array<std::string_view, 5> units{
+      "B", "KB", "MB", "GB", "TB"};
+    uint64_t rest = bytes;
+    auto unit = 0U;
+    for (; rest >= 1024U && unit < (units.size() - 1); ++unit) {
+        rest = (rest >> 10U);
+    }
+
+    uint64_t single = ((uint64_t)1 << (10U * unit));
+    double d_bytes = 1.0 + (double)(bytes - single) / single;
```

Comment:
should we avoid the FLOP in case this gets used in some hot path?

Reply:
we shouldn't use this at all, we have a different algorithm already implemented.